cvmfs/cache.cc

```diff
@@ -194,10 +194,13 @@ int CacheManager::RestoreState(const int fd_progress, void *data) {
       SendMsg2Socket(fd_progress, "unsupported state version!\n");
     abort();
   }
-  if (state->manager_type != id()) {
-    if (fd_progress >= 0)
-      SendMsg2Socket(fd_progress, "switching cache manager unsupported!\n");
-    abort();
+  // exception: we do allow switching from the non-refcounted
+  // to the refcounted posix cache manager
+  if (IsReloadCompatible(state->manager_type, id())) {
```

Comment:
```suggestion
  if (!IsReloadCompatible(state->manager_type, id())) {
```

We should also add a comment before on the test of the state version.  It probably works because both the regular and the refcounted manager use version 0. Now that we can switch between them, we have entangled their state versions.